pkg/controller/storagecluster/storagecluster.go

```diff
@@ -441,13 +437,49 @@ func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
 			logrus.WithError(err).Errorf("permission check for eks cloud drive failed")
 		}
 	}
-
 	// TODO: validate cloud permission for other providers as well
 
+	// Create StorageNodes to return pre-flight checks used by c.Driver.Validate()
+	k8sNodeList := &v1.NodeList{}
+	err = c.client.List(context.TODO(), k8sNodeList)
+	if err == nil {
+		for _, node := range k8sNodeList.Items {
+			logrus.Infof("Create pre-flight storage node entry for node: %s", node.Name)
+			c.createStorageNode(cluster, node.Name)
+		}
+	} else {
+		logrus.WithError(err).Errorf("Failed to get cluster nodes")
+	}
+
+	// Run driver specific pre-flights
+	if err = c.Driver.Validate(toUpdate); err != nil {
+		logrus.WithError(err).Errorf("pre-flight validate failed")
+	}
+
+	// Delete StorageNodes created for c.Driver.Validate() checks.
+	storageNodes := &corev1.StorageNodeList{}
+	err = c.client.List(context.TODO(), storageNodes,
+		&client.ListOptions{Namespace: toUpdate.Namespace})
+	if err == nil {
+		for _, storageNode := range storageNodes.Items {
+			logrus.Infof("Delete validate() storage node entry for node: %s", storageNode.Name)
+			err = c.client.Delete(context.TODO(), storageNode.DeepCopy())
+			if err != nil {
+				logrus.WithError(err).Errorf("failed to delete storage node entry %s: %v", storageNode.Name, err)
+			}
+		}
+	} else {
+		logrus.WithError(err).Errorf("Failed to get StorageNodes used for validate.")
+	}
+
 	condition := &corev1.ClusterCondition{
 		Source: pxutil.PortworxComponentName,
 		Type:   corev1.ClusterConditionTypePreflight,
 	}
+
+	// XXX - test
+	//err = fmt.Errorf("pre-flighr check force failure")
+
 	if err != nil {
```

Comment:
We would be capturing the wrong error here in case deleting storagenodes fail here. Also, if the cloud drive permission pre-check fails on line 436, we will overwrite that error.

Reply:
yeah thx I noticed that while I was debugging UTs issues.    I have that change coming in PRs which fixes the driver UT.   Basically when I cleanup the storageNodes used for pre-flight I am using a different err.